pkg/config/config_parser.go

```diff
@@ -150,22 +154,43 @@ func decodeConfig(r io.Reader) (*Config, diag.Diagnostics) {
 		return nil, diags
 	}
 
+	diags := diag.Diagnostics{}.Add(validateExtraKeys(yc.ExtraKeys))
+	diags = diags.Add(validateExtraKeys(yc.CloudQuery.ExtraKeys))
+
 	providers := yc.Providers
-	diags := diag.Diagnostics{}
 	for _, p := range providers {
 		p.Configuration, err = yaml.Marshal(p.ConfigKeys["configuration"])
 		if err != nil {
 			diags = diags.Add(diag.FromError(err, diag.INTERNAL, diag.WithSummary("ConfigKeys marshal failed")))
 			continue
 		}
+		delete(p.ConfigKeys, "configuration")
+		diags = diags.Add(validateExtraKeys(p.ConfigKeys))
+	}
+
+	if diags.HasErrors() {
+		return nil, diags
 	}
 
 	c := &Config{
 		CloudQuery: yc.CloudQuery,
 		Providers:  providers,
 	}
 
-	return c, nil
+	return c, diags
+}
+
+func validateExtraKeys(extraKeys map[string]interface{}) diag.Diagnostics {
+	if len(extraKeys) == 0 {
+		return nil
+	}
+
+	keys := sort.StringSlice(funk.Keys(extraKeys).([]string))
+	keyOrKeys := "keys"
+	if len(keys) == 1 {
+		keyOrKeys = "key"
+	}
+	return diag.FromError(errors.New("Encountered unknown element in config"), diag.USER, diag.WithSummary("Invalid configuration"), diag.WithDetails("Found unknown %s: %s", keyOrKeys, strings.Join(keys, ", ")))
```

Comment:
I don't think that this should be an error. I think it should be a warning. That way a newer config file can work with an older cq binary

Reply:
Not sure, maybe. Maybe different error levels depending on the block they're on (allow for adding a new block (e.g. `modules:`) in top level but don't allow for an extra key inside the `provider` etc.)

Reply:
I think kemal is correct here, this should be an error as the "config" is invalid, although only because of a whitespace perhaps, but it might have unintentional run for example if I wanted region X and Y if we get only a warning CQ will run and fetch all regions, which is not the intended action